tracing/trace.go

```diff
@@ -49,16 +51,41 @@ type trace struct {
 	tags     map[string]string
 }
 
+func (t *trace) internalTracer() *Tracer {
+	if t == nil {
+		return nil
+	}
+	if tracer, ok := t.tracer.(*Tracer); ok && tracer != nil {
+		return tracer
+	}
+	return nil
```

Comment:
don't return `nil` in-band, that's a recipe for nil pointer dereferences.  Return `*Tracer, bool`